### ISSUE TITLE
pylint should warn about lambda which use loop variables

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#176]
Originally reported by: **Mirko Dziadzka (BitBucket: [mirkodziadzka](http://bitbucket.org/mirkodziadzka), GitHub: @mirkodziadzka?)**

---

The following code should trigger a warning in pylint

```
#!python

def should_give_a_warning():
    res = []
    for i in range(10):
        res.append(lambda: i)
    return res
```

because the lambda does not bind the i from the loop but the last value of i the function. 

The result of this function is not was an ordinary python developer would expect here.

I'm not sure how to do this without triggering too many false positives. I would recommend a warning when lambda expression uses a outside variable  which can change after the lambda is created (assuming the lambda still exist at this point)

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/176
